src/configuration/applier/host.cc

```diff
@@ -83,10 +82,10 @@ void applier::host::add_object(
   config->hosts().insert(obj);
 
   // Create host.
-  com::centreon::engine::host*
-    h(add_host(
+  std::shared_ptr<com::centreon::engine::host> h =
+    std::make_shared<com::centreon::engine::host>(
         obj.host_id(),
-        obj.host_name().c_str(),
+        obj.host_name(),
         NULL_IF_EMPTY(obj.display_name()),
```

Comment:
if your constructor takes its string arguments as std::string, you could avoid this macro NULL_IF_EMPTY.

Reply:
done.